lib/adapter.js

```diff
@@ -0,0 +1,494 @@
+        var httpData = getHTTPData();
+		var probeData;
+		var heapData = getHeapData();
+		var usedHeap = getGCHeapData();
+		var messageData = getMessageData();
+
+		// Fire callback with data to match the strong-agent docs
+		callback("loop.count", eventLoopData.count);
+		callback("loop.minimum", eventLoopData.minimum);
+		callback("loop.maximum", eventLoopData.maximum);
+		callback("loop.average", eventLoopData.average);
+
+		callback("cpu.system", cpuData.system);
+		callback("cpu.total", cpuData.total);
+		callback("cpu.user", cpuData.user);
+
+		callback("http.connection.count", httpData.count);
+
+		// Callback with probe data of each probe if there have been events
+		for(i in probeList){
+			if (probeMetrics[probeList[i]]){
+				probeData = getProbeData(probeList[i]);
+				callback(probeList[i] + ".count", probeData.count);
+				callback(probeList[i] + ".average", probeData.average);
+				callback(probeList[i] + ".minimum", probeData.minimum);
+				callback(probeList[i] + ".maximum", probeData.maximum);
+			}
+		}
+
+		//if (usedHeap > 0) {
+			callback("gc.heap.used", usedHeap);
+		//}
+
+		callback("heap.total", heapData.total);
+		callback("heap.used", heapData.used);
+
+        if (messageData != null) {
+            callback("messages.out.count", messageData.sent);
+            callback("messages.in.count", messageData.received);  
+        }
+		
+		
+	}, callbackInterval);
+	setInterval(
+			function() {
+				if (objectTrackingEnabled) {
+					var histogram = appmetrics.getObjectHistogram();
+					var objectCounts = compareHistograms(lastObjectHistogram,
+							histogram);
+					lastObjectHistogram = histogram;
+					// get object stats
+					for ( var type in objectCounts) {
+						var tuple = objectCounts[type];
+						// Don't output events for no change? TODO - What's the
+						// correct
+						// behaviour here?
+						// TODO - Should we skip "weird" object types?
+						if (tuple['count'] != 0 && tuple['size'] != 0) {
+							callback("object." + type + ".count",
+									tuple['count']);
+							callback("object." + type + ".size", tuple['size']);
+						}
+					}
+					// fire events
+				}
+
+			}, objectTrackingCallbackInterval);
+}
+
+function start() {
+
+}
+
+function on() {
+	
+}
+
+function internal_on() {
+	
+}
+
+
+function profile(arg1, arg2, arg3) {
+
+}
+
+function configure(options) {
+
+}
+
+// CPU Profiling
+function startCpuProfiling() {
+	// If we aren't profiling already
+	if (!profiling) {
+		profiling = true;
+
+		// Instruct appmetrics to return JSON formatting and enable profiling
+		appmetrics.sendControlCommand("profiling_node",
+				"on,profiling_node_v8json");
+		appmetrics.enable('profiling');
+
+	}
+}
+
+function stopCpuProfiling(callback) {
+	if (profiling) {
+		
+		// Listen for JSON coming back from appmetrics
+		amapi.on('profiling', function(data) {
+			callback(JSON.stringify(data));
+		});
+
+		profiling = false;
+		appmetrics.disable('profiling');
```

Comment:
yes, 'disable' causes appmetrics to stop profiling and return all the data.  Its our wording